clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -589,12 +592,20 @@ public static Set<TopicPartition> generateRandomTopicPartitions(int numTopic, in
      * @return The caught exception cause
      */
     public static <T extends Throwable> T assertFutureThrows(Future<?> future, Class<T> exceptionCauseClass) {
-        ExecutionException exception = assertThrows(ExecutionException.class, future::get);
-        Throwable cause = exception.getCause();
-        assertEquals(exceptionCauseClass, cause.getClass(),
-            "Expected a " + exceptionCauseClass.getSimpleName() + " exception, but got " +
-                        cause.getClass().getSimpleName());
-        return exceptionCauseClass.cast(exception.getCause());
+        try {
+            future.get(DEFAULT_MAX_WAIT_MS, TimeUnit.MILLISECONDS);
+            fail("Should throw expected exception " + exceptionCauseClass.getSimpleName() + " but nothing was thrown.");
+        } catch (InterruptedException | ExecutionException | CancellationException e) {
+            Throwable cause = e instanceof ExecutionException ? e.getCause() : e;
+            assertInstanceOf(
+                exceptionCauseClass, cause,
+                "Expected " + exceptionCauseClass.getSimpleName() + ", but got " + cause
+            );
+            return exceptionCauseClass.cast(cause);
```

Comment:
I think we don't need an additional cast here.
`assertInstanceOf` has already done the job for us.

Reply:
I need to cast the `Throwable` to `T` class, thus we should `exceptionCauseClass.cast(cause);`